### PR TITLE
Introduce relative_anchor function.

### PR DESCRIPTION
**Pull request by floitsch**
On Windows a relative path can be either relative to the current working directory (or another file), or relative to the drive.

### CODE REVIEW

File: src/compiler/filesystem.h [hunk 1]
```diff
@@ -45,11 +45,25 @@ class Filesystem {
   virtual List<const char*> package_cache_paths() = 0;
 
   virtual bool is_absolute(const char* path) = 0;
+  // The path the non-absolute path is relative to.
+  // On Posix systems this is equal to `cwd`.
+  // On Windows, it can be `cwd`, or a drive (like "c:"), if the path starts with '\' or '/'.
+  virtual const char* relative_anchor(const char* path) {
+    return cwd();
```
Comment (kasperl) [inline src/compiler/filesystem.h:52]: As discussed, having these defined in the super class is a source of confusion. Maybe just drop them here?

File: src/compiler/filesystem_local_win.cc [hunk 2]
```diff
@@ -62,19 +62,33 @@ static int root_prefix_length(const char* path) {
...
     return 3;  // Drive root.
   }
   return 0;
 }
 
+const char* FilesystemLocal::relative_anchor(const char* path) {
+  ASSERT(!is_absolute(path));
+  if (path[0] == '/') {  // Safe to access even for empty strings.
+    // TODO(florian): error checking.
+    DWORD result_length = GetFullPathName("/", 0, NULL, NULL);
```
Comment (kasperl) [inline src/compiler/filesystem_local_win.cc:82]: Can this be cached in any meaningful way? Looks like a ton of allocations.

**Review by kasperl — APPROVED** [review]
Looks good to me.

File: src/compiler/filesystem_local_win.cc (hunk 2 above)
Comment (floitsch) [reply src/compiler/filesystem_local_win.cc:82]: I would like to revisit this area again.
The current setup (mostly?) fixes the immediate issues, but I hope to find a better API.

**Pull request merged**